api/invite.go

```diff
@@ -526,3 +851,357 @@ func (a *Api) SendInvite(res http.ResponseWriter, req *http.Request, vars map[st
+	var ib = &inviteBody{}
+	if err := json.NewDecoder(req.Body).Decode(ib); err != nil {
+		log.Printf("UpdateInvite: error decoding invite to detail %v\n", err)
+		statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_DECODING_INVITE)}
+		a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+		return
+	}
+
+	if ib.Email == "" || ib.TeamID == "" {
+		statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_MISSING_DATA_INVITE)}
+		a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+		return
+	}
+	if ib.IsAdmin == "" {
+		ib.IsAdmin = "false"
+	}
+
+	_, team, err := a.userIsTeamAdmin(tokenValue, token.UserID, ib.TeamID, res)
+	if err != nil {
+		return
+	}
+
+	var isAdmin = strings.ToLower(ib.IsAdmin) == "true"
+	if isAdmin {
+		invitedUsr := a.findExistingUser(ib.Email, a.sl.TokenProvide())
+		if invitedUsr != nil && invitedUsr.UserID != "" {
+			if admin, err := a.isTeamAdmin(invitedUsr.UserID, team); admin || err != nil {
+				statusErr := &status.StatusError{Status: status.NewStatus(http.StatusConflict, STATUS_ALREADY_ADMIN)}
+				a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+				return
+			}
+		}
+		invite, _ := models.NewConfirmationWithContext(
+			models.TypeMedicalTeamDoAdmin,
+			models.TemplateNameMedicalteamDoAdmin,
+			invitorID,
+			ib.Permissions)
+
+		// if the invitee is already a user, we can use his preferences
+		invite.TeamID = ib.TeamID
+		invite.Email = ib.Email
+		invite.IsAdmin = ib.IsAdmin
+		invite.Status = models.StatusCompleted
+		if invitedUsr != nil {
+			invite.UserId = invitedUsr.UserID
+			// does the invitee have a preferred language?
+			inviteeLanguage = a.getUserLanguage(invite.UserId, res)
+		}
+
+		var member = store.Member{
+			UserID:           invite.UserId,
+			TeamID:           ib.TeamID,
+			Role:             ib.IsAdmin,
+			InvitationStatus: "accepted",
+		}
+		if _, err := a.perms.UpdateTeamMember(tokenValue, member); err != nil {
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusInternalServerError, STATUS_ERR_UPDATING_TEAM)}
+			a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+			return
+		}
+
+		if a.addOrUpdateConfirmation(req.Context(), invite, res) {
+			a.logAudit(req, "invite created")
+
+			if err := a.addProfile(invite); err != nil {
+				log.Println("SendInvite: ", err.Error())
+			} else {
+
+				emailContent := map[string]interface{}{
+					"MedicalteamName": team.Name,
+					"Email":           invite.Email,
+					"Language":        inviteeLanguage,
+				}
+
+				if a.createAndSendNotification(req, invite, emailContent, inviteeLanguage) {
+					a.logAudit(req, "invite sent")
+				} else {
+					a.logAudit(req, "invite failed to be sent")
+					log.Print("Something happened generating an invite email")
+					res.WriteHeader(http.StatusUnprocessableEntity)
+					return
+				}
+			}
+			a.sendModelAsResWithStatus(res, invite, http.StatusOK)
+			return
+		}
```

Comment:
Else? !isAdmin ? Don't we want to handle this usecase when an admin removes admin rights to another admin?

Reply:
????